### PR TITLE
Add regex to detect higherkind annotation

### PR DESCRIPTION
**Pull request by rachelcarmena**
Pretty small change to detect `higherkind` annotation like:

```
@higherkind
```

or:

```
@arrow.higherkind
```

### CODE REVIEW

File: modules/meta/arrow-meta-prototype/compiler-plugin/src/main/kotlin/arrow/meta/plugins/higherkind/HigherKindPlugin.kt [hunk 1]
```diff
@@ -75,7 +75,7 @@ private val KtClass.partialKindAritySuffix: String
   get() = (arity - 1).let { if (it > 1) "$it" else "" }
 
 fun isHigherKindedType(ktClass: KtClass): Boolean =
-  ktClass.annotationEntries.any { it.text == "@higherkind" } &&
+  ktClass.annotationEntries.any { it.text.matches(Regex("@(arrow.)?higherkind")) } &&
```
Comment (pakoito) [inline modules/meta/arrow-meta-prototype/compiler-plugin/src/main/kotlin/arrow/meta/plugins/higherkind/HigherKindPlugin.kt]: `\.`?

Comment (rachelcarmena) [reply modules/meta/arrow-meta-prototype/compiler-plugin/src/main/kotlin/arrow/meta/plugins/higherkind/HigherKindPlugin.kt]: Good point, @pakoito!! Thanks!! Solution: `\\.`

```
>>> val regex = Regex("@(arrow.)?higherkind")
>>> "@higherkind".matches(regex)
res1: kotlin.Boolean = true
>>> "@higherkind2".matches(regex)
res2: kotlin.Boolean = false
>>>  "@.higherkind".matches(regex)
res3: kotlin.Boolean = false
>>> "@arrow.higherkind".matches(regex)
res4: kotlin.Boolean = true
>>> "@arrow-higherkind".matches(regex)
res5: kotlin.Boolean = true
>>> 
```

``` 
>>> val regex = Regex("@(arrow\\.)?higherkind")
>>> "@higherkind".matches(regex)
res1: kotlin.Boolean = true
>>> "@higherkind2".matches(regex)
res2: kotlin.Boolean = false
>>> "@.higherkind".matches(regex)
res3: kotlin.Boolean = false
>>> "@arrow.higherkind".matches(regex)
res4: kotlin.Boolean = true
>>> "@arrow-higherkind".matches(regex)
res5: kotlin.Boolean = false
>>> 
```

Comment (rachelcarmena) [reply modules/meta/arrow-meta-prototype/compiler-plugin/src/main/kotlin/arrow/meta/plugins/higherkind/HigherKindPlugin.kt]: Fixed!! 
However, the target of this PR isn't `master`. 
I will be happy with your approval :wink:

Comment (pakoito) [reply modules/meta/arrow-meta-prototype/compiler-plugin/src/main/kotlin/arrow/meta/plugins/higherkind/HigherKindPlugin.kt]: Go for it!

**Pull request merged**